xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -314,9 +323,22 @@ public Document configToDocument(Configuration configuration) throws JAXBExcepti
 
   public static class FatalErrorHandler implements ErrorHandler {
 
-    private static final Pattern ABSTRACT_TYPE_FAILURES = of("service-creation-configuration", "service-configuration", "resource")
-      .map(element -> quote(format("\"http://www.ehcache.org/v3\":%s", element)))
-      .collect(collectingAndThen(joining("|", "^\\Qcvc-complex-type.2.4.a\\E.*'\\{.*(?:", ").*\\}'.*$"), Pattern::compile));
+    private static final Collection<Pattern> ABSTRACT_TYPE_FAILURES;
+    static {
+      ObjectFactory objectFactory = new ObjectFactory();
+      List<QName> abstractTypes = asList(
+        objectFactory.createServiceCreationConfiguration(null).getName(),
+        objectFactory.createServiceConfiguration(null).getName(),
+        objectFactory.createResource(null).getName());
+
+      ABSTRACT_TYPE_FAILURES = asList(
+        //Xerces
+        abstractTypes.stream().map(element -> quote(format("\"%s\":%s", element.getNamespaceURI(), element.getLocalPart())))
+          .collect(collectingAndThen(joining("|", "^\\Qcvc-complex-type.2.4.a\\E.*'\\{.*(?:", ").*\\}'.*$"), Pattern::compile)),
+        //Saxon
+        abstractTypes.stream().map(element -> quote(element.getLocalPart()))
+          .collect(collectingAndThen(joining("|", "^.*\\QThe content model does not allow element\\E.*(?:", ").*"), Pattern::compile)));
```

Comment:
The `SAXParseException` message is _stable_ regardless of Locale?

Reply:
It almost certainly isn't... but there is no way to handle it. Xerces is nice in that it quotes the relevant section of the spec which is probably stable. What saves us here is that this condition doesn't affect pass/fail behavior it just affects how helpful we can be when we see this failures.